PickMeUp/PickMeUp/Model/PMDataManager.m

```diff
@@ -0,0 +1,56 @@
+//
+//  PMDataManager.m
+//  PickMeUp
+//
+//  Created by Abigail Shilts on 7/27/22.
+//
+#import "Parse/Parse.h"
+#import "PMCachingFunctions.h"
+#import "PMDataManager.h"
+#import "PMTree.h"
+#import "StringsList.h"
+
+@interface PMDataManager ()
+@property (nonatomic, copy) void (^block)(NSArray<PMConversation *> *);
+@end
+
+@implementation PMDataManager
+
+-(void)fillConversations:(void(^)(NSArray<PMConversation *> *))block {
+    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void){
+        self.block = block;
+        self.conversations = [PMCachingFunctions retreiveConversationCache];
+        dispatch_async(dispatch_get_main_queue(), ^(void){
+            self.block(self.conversations);
```

Comment:
it's a good practice to check if block is before calling. If someone passes nil by mistake the app will crash.